tools/rostest/src/rostest/rostest_parent.py

```diff
@@ -48,14 +49,37 @@
 
 class ROSTestLaunchParent(roslaunch.parent.ROSLaunchParent):
 
-    def __init__(self, config, roslaunch_files, port=0):
+    def __init__(self, config, roslaunch_files, port=0, local=False, clear=False):
         if config is None:
             raise Exception("config not initialized")
         # we generate a run_id for each test
-        run_id = roslaunch.core.generate_run_id()
+        if local:
+            param_server = rosgraph.Master('/roslaunch')
+            have_master = False
+            try:
+                run_id = param_server.getParam('/run_id')
+                have_master = True
+            except:
+                run_id = roslaunch.core.generate_run_id()
```

Comment:
This is copied nearly verbatim from roslaunch, so I'm not really sure which exceptions to expect. Would a catch-all and print be more reasonable?

https://github.com/ros/ros_comm/blob/78aa1c4ec879668f686aba873f52968b1bd7aa73/tools/roslaunch/src/roslaunch/rlutil.py#L173-L181